cmd/tracee-ebpf/capabilities.go

```diff
@@ -73,16 +74,47 @@ func getCapabilitiesRequiredByTraceeEvents(cfg *tracee.Config) []cap.Value {
 	return removeDupCaps(caps)
 }
 
+// Retrieve the value of the kernel parameter perf_event_paranoid
+func getKernelPerfEventParanoidValue() int {
+	// Read the kernel parameter perf_event_paranoid
+	value, err := os.ReadFile("/proc/sys/kernel/perf_event_paranoid")
+	// If file read failed, returns the most restrictive value
+	if err != nil {
+		return 4
+	}
+	// Strip the newline character
+	value = value[:len(value)-1]
+	// Convert the value to an integer
+	intVal, err := strconv.ParseInt(string(value), 0, 16)
+	// If conversion failed, returns the most restrictive value
+	if err != nil {
+		return 4
+	}
+	return int(intVal)
+}
+
 // Get all capabilities required for eBPF usage (including perf buffers maps management)
 func getCapabilitiesRequiredByEBPF(OSInfo KernelVersionInfo) ([]cap.Value, error) {
 	// In kernel 5.8, CAP_BPF and CAP_PERFMON capabilities were introduced in order to replace CAP_SYS_ADMIN when
 	// loading eBPF programs.
-	// For some reasons, some distributions using new kernels still need CAP_SYS_ADMIN,
-	// so tracee still use it instead of the new capabilities.
-	caps := []cap.Value{
-		cap.IPC_LOCK,
-		cap.SYS_RESOURCE,
-		cap.SYS_ADMIN,
+	kernelParanoidValue := getKernelPerfEventParanoidValue()
+	var caps []cap.Value
+	// if kernelParanoidValue is too high, CAP_SYS_ADMIN is required
+	if kernelParanoidValue > 1 {
```

Comment:
This one is weird.
You wrote in the issue that you saw that values up to 3 enable the use of `CAP_BPF` and `CAP_PERFMON`.
Why did you change it to be 1?
In my system I have the value of 2, though it works for me with  `CAP_BPF` and `CAP_PERFMON`. Is this a bug?

Reply:
Just wanted to be safe as the documentation mentions  `2 - disallow kernel profiling for unpriv`. But it does work qith values up to 3 on my machine.

Reply:
I think we can have the assumption that tracee run as a privileged process and user.